packer/vault-policies/write-node-policies.py

```diff
@@ -0,0 +1,84 @@
+#!/usr/bin/env python
+
+import json
+import sys
+
+NETWORK_ID = ''
+AWS_ACCOUNT_ID = ''
+SETUP_FILE = ''
+OUTPUT_DIR = ''
+
+def policy_string(region, role, role_index, cluster_index):
+    if role == "bootnodes":
+        return """\
+path \\"quorum/bootnodes/passwords/{0}/{1}\\" {{
+    capabilities = [\\"create\\", \\"update\\", \\"read\\"]
+}}
+
+path \\"quorum/bootnodes/keys/{0}/{1}\\" {{
+    capabilities = [\\"create\\", \\"update\\", \\"read\\"]
+}}
+
+path \\"quorum/bootnodes/addresses/{0}/{1}\\" {{
+    capabilities = [\\"create\\", \\"update\\", \\"read\\"]
+}}
+""".format(region, role_index)
+    else:
+        return """\
+path \\"quorum/passwords/{0}/{1}\\" {{
+    capabilities = [\\"create\\", \\"update\\", \\"read\\"]
+}}
+
+path \\"quorum/keys/{0}/{1}\\" {{
+    capabilities = [\\"create\\", \\"update\\", \\"read\\"]
+}}
+
+path \\"quorum/addresses/{0}/{1}\\" {{
+    capabilities = [\\"create\\", \\"update\\", \\"read\\"]
+}}
+
+path \\"quorum/{2}/{0}/{3}\\" {{
+    capabilities = [\\"create\\", \\"update\\", \\"read\\"]
+}}
+""".format(region, cluster_index, role, role_index)
+
+def write_all_role_nodes(region, role, node_count, start_index):
+    for (role_index, cluster_index) in zip(range(node_count),range(start_index, start_index+node_count)):
+        node_rolename = "quorum-{0}-network-{1}-{2}-{3}".format(region, NETWORK_ID, role, role_index)
+        with open(SETUP_FILE, "a") as f:
+            policy_text = policy_string(region, role, role_index, cluster_index)
+            f.write("\necho \"{1}\" | vault policy-write {0} - \n".format(node_rolename, policy_text))
+            f.write("vault write auth/aws/role/{0} auth_type=iam policies=\"base-read,{0}\" bound_iam_principal_arn=arn:aws:iam::{1}:role/{0} || true \n".format(node_rolename, AWS_ACCOUNT_ID))
+
+if __name__ == "__main__":
+    (
```

Comment:
If it doesn't have enough arguments, then it vomits saying that it needs more args -- it probably doesn't barf on too many args, though, I'll add something for that.

I'm thinking that since the script is only called via script and can only be called in one way with all the right args, argparse is overkill.  Happy to build that code in there if you think it's important, but this tuple unpacking gets my specific job done.

Reply:
This is fine as long as we make it fail loudly when we get too many args